pkg/scheduler/scheduler.go

```diff
@@ -422,9 +422,6 @@ func (sched *Scheduler) finishBinding(prof *profile.Profile, assumed *v1.Pod, ta
 	}
 	if err != nil {
 		klog.V(1).Infof("Failed to bind pod: %v/%v", assumed.Namespace, assumed.Name)
-		if err := sched.SchedulerCache.ForgetPod(assumed); err != nil {
-			klog.Errorf("scheduler cache ForgetPod failed: %v", err)
-		}
```

Comment:
Good removal. This ensures we run `ForgetPod()` once, and ensure it's called at the end the failure path instead of `defer finishBinding()` in `bind()`.